include/dpp/sync.h

```diff
@@ -27,54 +27,55 @@
-     * @tparam T type of expected return value, should match up with the method called
-     * @tparam F Type of class method in dpp::cluster to call.
-     * @tparam Ts Function parameters in method call
-     * @param c A pointer to dpp::cluster object
-     * @param func pointer to class method in dpp::cluster to call. This can call any
-     * dpp::cluster member function who's last parameter is a dpp::command_completion_event_t callback type.
-     * @param args Zero or more arguments for the method call
-     * @return An instantiated object of type T
-     * @throw dpp::rest_exception On failure of the method call, an exception is thrown
-     */
-    template<typename T, class F, class... Ts> T sync(class cluster* c, F func, Ts&&... args) {
-        std::promise<T> _p;
-        std::future<T> _f = _p.get_future();
-        /* (obj ->* func) is the obscure syntax for calling a method pointer on an object instance */
-        (c ->* func)(std::forward<Ts>(args)..., [&_p](const auto& cc) {
-            try {
-                if (cc.is_error()) {
-                    throw dpp::rest_exception(cc.get_error().message);
-                } else {
-                    try {
-                        _p.set_value(std::get<T>(cc.value));
-                    } catch (const std::exception& e) {
-                        throw dpp::rest_exception(e.what());
-                    }
-                }
-            } catch (const dpp::rest_exception&) {
-                _p.set_exception(std::current_exception());
-            }
-        });
+	/**
+	 * @brief Call a D++ REST function synchronously.
+	 *
+	 * Synchronously calling a REST function means *IT WILL BLOCK* - This is a Bad Thing™ and strongly discouraged.
+	 * There are very few circumstances you actually need this. If you do need to use this, you'll know it.
+	 *
+	 * Example:
+	 *
+	 * ```cpp
+	 * dpp::message m = dpp::sync<dpp::message>(&bot, &dpp::cluster::message_create, dpp::message(channel_id, "moo."));
+	 * ```
+	 *
+	 * @warning As previously mentioned, this template will block. It is ill-advised to call this outside of
+	 * a separate thread and this should never be directly used in any event such as dpp::cluster::on_interaction_create!
+	 * @tparam T type of expected return value, should match up with the method called
+	 * @tparam F Type of class method in dpp::cluster to call.
+	 * @tparam Ts Function parameters in method call
+	 * @param c A pointer to dpp::cluster object
+	 * @param func pointer to class method in dpp::cluster to call. This can call any
+	 * dpp::cluster member function who's last parameter is a dpp::command_completion_event_t callback type.
+	 * @param args Zero or more arguments for the method call
+	 * @return An instantiated object of type T
+	 * @throw dpp::rest_exception On failure of the method call, an exception is thrown
+	 */
+	template<typename T, class F, class... Ts> T sync(class cluster* c, F func, Ts&&... args) {
```

Comment:
This shouldn't be tab-indented as it's a namespace, https://discord.com/channels/825407338755653642/825411104208977952/1159604805979881472